### PR TITLE
Migrate module configuration in RecoMET to use default cfipython

### PR DESCRIPTION
**Pull request by jeongeun**
#### PR description: 
Optimization of the python configurations: Improve maintainability by cleaning up the duplicated and cloning from the default/reference configurations. (The previous PRs : #33207, #33307, #33352, #33543, #33563, #33671, #34007, #34091, #34009, #34155, #34371, #34480, #34543 )
 
In this PR, 1 file changed.  
        modified:   RecoMET/METPUSubtraction/python/mvaPFMET_leptons_cfi.py

Update
- Replace explicit configuration with a reference from cfipython/. (migrating EDProducer("type", .. -> typeDefaylt.clone() ) 
- Remove the type specifications already presented in cfipython/fillDescriptions reference for improved syntax safety.
- Remove the duplicated parameters that are exactly the same value in cfipython reference.

#### PR validation:
Event Content comparison check was also done and there is no change with these updates.
Tested in CMSSW_12_0_X, the basic test all passed in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html).

### CODE REVIEW

**Comment by perrotta** [comment]
please test

File: RecoMET/METPUSubtraction/python/mvaPFMET_leptons_cfi.py [hunk 1]
```diff
@@ -4,100 +4,98 @@
...
+import RecoTauTag.RecoTau.pfRecoTauDiscriminationAgainstMuon2_cfi as _mod
 
-hpsPFTauDiscriminationAgainstMuon2 = cms.EDProducer("PFRecoTauDiscriminationAgainstMuon2",
-                                                         PFTauProducer = cms.InputTag('hpsPFTauProducer'),
-                                                         Prediscriminants = requireDecayMode.clone(),
-                                                         discriminatorOption = cms.string('loose'), # available options are: 'loose', 'medium', 'tight'
-                                                         HoPMin = cms.double(0.2)
-                                                     )
-
+PFTauDiscriminationAgainstMuon2 = _mod.pfRecoTauDiscriminationAgainstMuon2.clone(
```
Comment (perrotta) [inline RecoMET/METPUSubtraction/python/mvaPFMET_leptons_cfi.py:69]: The name of the module was `hpsPFTauDiscriminationAgainstMuon2`

File: RecoMET/METPUSubtraction/python/mvaPFMET_leptons_cfi.py [hunk 2]
```diff
@@ -4,100 +4,98 @@
...
+    PFTauProducer = 'hpsPFTauProducer',
+    Prediscriminants = requireDecayMode.clone(),
+    discriminatorOption = 'loose', # available options are: 'loose', 'medium', 'tight'
+)
 
 hpsPFTauDiscriminationByMVAIsolation = cms.EDProducer(
     "PFRecoTauDiscriminationByMVAIsolation",
```
Comment (perrotta) [inline RecoMET/METPUSubtraction/python/mvaPFMET_leptons_cfi.py:76]: I can't find this plugin, while I see `PFRecoTauDiscriminationByMVAIsolationRun2` instead
@mbluj @swozniewski is it just me, or this configuration is outdated by now?

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by perrotta** [comment]
@camclean @alefisico , perhaps the question about the update status of the configurations in RecoMET/METPUSubtraction/python/mvaPFMET_leptons_cfi.py is better addressed to you than to the tau contacts.
My suspect is that either that config is never used since a while already and can be removed, or it has to be updated in order to configure (at least) exhisting plugins.
Could you please have a look and provide your opinion?

**Comment by perrotta** [comment]
> @camclean @alefisico , perhaps the question about the update status of the configurations in RecoMET/METPUSubtraction/python/mvaPFMET_leptons_cfi.py is better addressed to you than to the tau contacts.
> My suspect is that either that config is never used since a while already and can be removed, or it has to be updated in order to configure (at least) exhisting plugins.
> Could you please have a look and provide your opinion?

I understand that in summer it can be hard to find people around... Let try with the JetMet conveners now: @lathomas @kirschen

**Comment by slava77** [comment]
> > @camclean @alefisico , perhaps the question about the update status of the configurations in RecoMET/METPUSubtraction/python/mvaPFMET_leptons_cfi.py is better addressed to you than to the tau contacts.
> > My suspect is that either that config is never used since a while already and can be removed, or it has to be updated in order to configure (at least) exhisting plugins.
> > Could you please have a look and provide your opinion?
> 
> I understand that in summer it can be hard to find people around... Let try with the JetMet conveners now: @lathomas @kirschen

[a weekly reminder]

**Comment by jpata** [comment]
I opened an issue for the dead code https://github.com/cms-sw/cmssw/issues/34839 so we can meanwhile move forward with this.

**Comment by jpata** [comment]
+reconstruction
- technical, migrates python configs
- no differences were observed in the tests

**Comment by perrotta** [comment]
+1
- Succesfully tested 16 days ago, but sit is a simple update, and quite likely the affected config is not even used (see #34839)

**Pull request merged**